src/data/Data.js

```diff
@@ -147,7 +147,11 @@ Data.prototype = {
 
 		// Remove the expando if there's no more data
 		if ( key === undefined || jQuery.isEmptyObject( cache ) ) {
-			delete owner[ this.expando ];
+			// Webkit & Blink performance suffers when deleting properties
```

Comment:
I guess this should be a "Support:" comment, we didn't do it for the `compat` but perhaps we should. 

So we talking about Safari, Chrome. Could try to identify affected versions? I guess we can omit other Chromium browsers since we don't do it in other cases.